frame/wormhole/backing/s2s/src/lib.rs

```diff
@@ -129,14 +139,63 @@ pub mod pallet {
 		NonceDuplicated,
 	}
 
+	/// Period between security limitation.
+	#[pallet::storage]
+	#[pallet::getter(fn secure_limited_period)]
+	pub type SecureLimitedPeriod<T> = StorageValue<_, BlockNumberFor<T>, ValueQuery>;
+
+	/// `(Spent, Maximum)` amount of *RING* security limitation each [`LimitedPeriod`].
+	#[pallet::storage]
+	#[pallet::getter(fn secure_limited_ring_amount)]
+	pub type SecureLimitedRingAmount<T> =
+		StorageValue<_, (RingBalance<T>, RingBalance<T>), ValueQuery>;
+
 	#[pallet::storage]
 	#[pallet::getter(fn locked_queue)]
 	pub type LockedQueue<T: Config> =
 		StorageMap<_, Identity, TokenMessageId, (AccountId<T>, Token), ValueQuery>;
 
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub secure_limited_period: BlockNumberFor<T>,
+		pub secure_limited_ring_amount: RingBalance<T>,
+	}
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				secure_limited_period: Zero::zero(),
+				secure_limited_ring_amount: Zero::zero(),
+			}
+		}
+	}
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			<SecureLimitedPeriod<T>>::put(self.secure_limited_period);
+			<SecureLimitedRingAmount<T>>::put((
+				<RingBalance<T>>::zero(),
+				self.secure_limited_ring_amount,
+			));
+		}
+	}
+
 	#[pallet::pallet]
 	pub struct Pallet<T>(_);
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(now: BlockNumberFor<T>) -> Weight {
+			if (now % <SecureLimitedPeriod<T>>::get()).is_zero() {
```

Comment:
Slight different with 
https://github.com/darwinia-network/darwinia-bridge-sol/blob/3d826353655d3695e62028487b2331ad2a7123ad/contracts/utils/contracts/DailyLimit.sol#L53

I think it is ok. Smart contract require tx to trigger the time update, we do not need this.

Reply:
Will now%0 be panic? better to omit the cases of  `<SecureLimitedPeriod<T>>::get()` is zero?